### PR TITLE
Folivora97/20240416 1

### PR DESCRIPTION
**Pull request by folivora97**

### CODE REVIEW

File: app/classes/DB.php [hunk 1]
```diff
@@ -39,33 +40,35 @@ class DB {
     /**
      * Constructor of the DB class. Calls the methods to load environment variables and establish a connection to the database.
      */
-    public function __construct(bool $autoconnect = true, bool $autoclose = false) {
+    public function __construct(bool $autoConnect = true, private readonly bool $autoClose = false)
```
Comment (CallMeLeon167) [inline app/classes/DB.php:43]: readonly is > PHP 8.1, we can't use this

File: app/classes/DB.php [hunk 2]
```diff
@@ -39,33 +40,35 @@ class DB {
...
-    
+
     /**
      * Establishes a connection to the database.
      *
      * @param string $host The database host.
      * @param string $user The database username.
      * @param string $pass The database password.
-     * @param string $dbname The database name.
+     * @param string $dbName
```
Comment (CallMeLeon167) [inline app/classes/DB.php]: Where is the comment of the param 😄

File: app/classes/DB.php [hunk 3]
```diff
@@ -39,33 +40,35 @@ class DB {
...
      */
-    public function connect(string $host, string $user, string $pass, string $dbname) {
-        $this->conn = @new \mysqli($host, $user, $pass, $dbname);
+    public function connect(string $host, string $user, string $pass, string $dbName): void
```
Comment (CallMeLeon167) [inline app/classes/DB.php:67]: I don't like to return void. If there is no return type, it's always void :). I mean, we can leave it there, but just saying.

File: app/classes/DB.php [hunk 4]
```diff
@@ -79,17 +82,19 @@ public function connect(string $host, string $user, string $pass, string $dbname
...
         }
-        $this->connect($host, $user, $pass, $dbname);
+        $this->connect($host, $user, $pass, $dbName);
     }
 
     /**
      * Restores the default database connection and closes the current connection if it's active.
      */
-    public function defaultConnection() {
+    public function defaultConnection(): void
```
Comment (CallMeLeon167) [inline app/classes/DB.php:96]: same here with void

File: app/classes/DB.php [hunk 5]
```diff
@@ -252,25 +265,33 @@ public function sql2db(string $query, array $params = []) {
      *
      * @param string $query The SQL query.
      * @param array $params Parameters for the SQL query (optional).
-     * @return string The result of the SQL query as a JSON-encoded string.
+     * @return string|bool The result of the SQL query as a JSON-encoded string.
+     * @throws Exception
      */
-    public function sql2json(string $query, array $params = []):string {
+    public function sql2json(string $query, array $params = []): string|bool
```
Comment (CallMeLeon167) [inline app/classes/DB.php:271]: Same here, we can't use "string|bool". Union Types is an > PHP 8 thing. PHP 7 only supports 1 return type.

File: app/classes/DB.php [hunk 6]
```diff
@@ -252,25 +265,33 @@ public function sql2db(string $query, array $params = []) {
...
         }
+
+        return false; //temporary stopgap
     }
 
     /**
      * Executes an SQL query from a file and returns the result as JSON.
      *
-     * @param string $filename The filename of the SQL query.
+     * @param string $fileName
```
Comment (CallMeLeon167) [inline app/classes/DB.php]: Again, the comment is despawning :)

File: app/classes/DB.php [hunk 7]
```diff
@@ -252,25 +265,33 @@ public function sql2db(string $query, array $params = []) {
...
      * @param array $params Parameters for the SQL query (optional).
-     * @return string The result of the SQL query as a JSON-encoded string.
+     * @return string|bool The result of the SQL query as a JSON-encoded string.
+     * @throws Exception
      */
-    public function sql2json_file(string $filename, array $params = []):string {
-        if (!empty($filename)) {
-            return json_encode($this->sql2array_file($filename, $params));
+    public function sql2jsonFile(string $fileName, array $params = []): string|bool
```
Comment (CallMeLeon167) [inline app/classes/DB.php:288]: Union Types again

File: app/classes/DB.php [hunk 8]
```diff
@@ -406,23 +431,29 @@ protected function checkDataTable(){
...
     /**
      * Retrieves data from the database based on the given name.
      *
      * @param string $name The name of the data to retrieve.
      * @return mixed The value of the data if found, or false if not found.
+     *
+     * @throws Exception
      */
-    public function getData(string $name){
+    public function getData(string $name): mixed
```
Comment (CallMeLeon167) [inline app/classes/DB.php:455]: The return type "mixed" is like the void, I don't like to type this to the function. But as I said, we can leave this if your happy with it

File: app/classes/DB.php (hunk 1 above)
Comment (folivora97) [reply app/classes/DB.php:43]: Requirement will change to >= 8.1

**Review by CallMeLeon167 — APPROVED** [review]
Looks very crispy 🔥

**Pull request merged**